cleanup_env.py

```diff
@@ -0,0 +1,264 @@
+        return
+
+    user_ = OSP_IDENTITY.get_user(node.extra["userId"])
+    if node.state.lower() != "error":
+        node_age = datetime.now(timezone.utc) - node.created_at
+        max_age = 3 if tenant == "ceph-jenkins" else 14
+
+        if (max_age - 1) > node_age.days > (max_age // 2):
+            if user_.name == "psi-ceph-jenkins":
+                return
+
+            add_key_to_ref(results, "warn", user_.email, node.name, tenant)
+            return
+
+        if node_age.days < max_age:
+            return
+    else:
+        add_key_to_ref(results, "error", user_.email, node.name, tenant)
+
+    try:
+        ceph_node = CephVMNodeV2(
+            username=osp_cred["username"],
+            password=osp_cred["password"],
+            auth_url=osp_cred["auth-url"],
+            auth_version=osp_cred["auth-version"],
+            tenant_name=tenant,
+            tenant_domain_id=osp_cred["tenant-domain-id"],
+            service_region=osp_cred["service-region"],
+            domain_name=osp_cred["domain"],
+            node_name=node.name,
+        )
+
+        ceph_node.delete()
+        if user_.name == "psi-ceph-jenkins":
+            # PSI Ceph Jenkins is a service account. Hence, ignoring email
+            return
+
+        add_key_to_ref(results, "deleted", user_.email, node.name, tenant)
+    except BaseException:  # noqa
+        add_key_to_ref(results, "marked", user_.email, node.name, tenant)
+
+
+def send_email(payload: Dict) -> list:
+    """
+    Sends an email to all recipients given in the payload.
+
+    Arguments:
+        payload     A dictionary having the primary key as email id
+
+    Returns:
+        list
+
+    payload example
+        {
+            "abc@xyz.com": {
+                                "warn": { "ceph-ci": ["vm1", "vm2"] },
+                                "deleted": { "ceph-jenkins": ["vm3"] },
+                                "marked": { "ceph-core": ["vm10"] }
+                            },
+            "abd@xyz.com": {
+                                "warn": { "ceph-jenkins": [...]},
+                            }}
+    """
+    failed_vm = []
+    sender = "cephci@redhat.com"
+    for email, ct in payload.items():
+        if email is not None:
+            recipient = email
+
+            msg = MIMEMultipart("alternative")
+            msg[
+                "Subject"
+            ] = "Notification: Your subscriptions are marked/removed in RHOS-D"
+            msg["From"] = sender
+            msg["To"] = recipient
+            html = """\
+                    <html><head><style>table, th, td {border: 1px solid black; }</style>
+                        </head><body>
+                            <p>Hi,<br><br>
+                            Below cloud subscriptions are expired/expiring.<br>
+                            The durations for subscriptions expire in project are for ceph-jenkins its 3 days,
+                            for ceph-ci and ceph-core its 2 weeks i.e, 14days.</p>"""
```

Comment:
try to use a template if you can.

Reply:
Will take this as an enhancement in next Iteration.